repose-aggregator/src/docs/asciidoc/filters/api-validator.adoc

```diff
@@ -11,45 +26,283 @@ include::../_includes/in-progress.adoc[]
 
 == Prerequisites & Postconditions
 === Required headers
-???.
+This filter does not require any request headers.
+
+However, the configuration can require specific headers and/or values be present in order to successfully exit this filter.
+The most common use of a request header by this filter is for <<../recipes/role-based-access-control.adoc#,Role-Based Access Control (RBAC)>> using the `X-Roles` header populated by the <<../filters/keystone-v2.adoc#,Keystone v2 filter>>.
 
 === Required preceding filters
-???.
+This filter has no dependencies on other filters and can be placed wherever it is needed in the filter chain.
+
+However, due to the nature of this filter it is typically placed early in the filter chain immediately after any authentication filters.
 
 === Request headers created
-???.
+This filter only creates/modifies request headers when configured for <<../recipes/delegation.adoc#,Delegation>>.
+If delegation is enabled, then the `X-Delegated` header is created.
+This is mainly intended for use by the <<herp.adoc#, Highly Efficient Record Processor (HERP) filter>> and <<derp.adoc#, Delegation Response Processor (DeRP) filter>> for internal delegation processing within Repose.
+However, it can be exposed to the origin service under certain configurations.
 
 === Request body changes
-???.
+This filter does not modify the response body.
 
 === Recommended follow-on (succeeding) filters
-???.
+This filter is not strictly required by any other filters.
 
 === Response body changes
-???.
+This filter does not modify the response body.
 
 === Response headers created
-???.
+This filter does not create/modify any response headers.
 
 === Response Status Codes
-???.
+[cols="2", options="header,autowidth"]
+.Status Codes
+|===
+| Status Code
+| Reasons
+
+| _Unauthorized_ (`401`)
+| A requested resource or method requires a specific `X-Authenticated-By` request header value that was not found.
+
+| _Forbidden_ (`403`)
+| A requested resource or method requires a specific `X-Roles` request header value that was not found.
+
+| _Not Found_ (`404`)
+| The filter determined that the URI is invalid.
+
+  When `mask-rax-roles-403` attribute is enabled (i.e. `true`), this could also indicate the supplied `X-Roles` request header value did not have access to the URI.
+
+| _Method Not Allowed_ (`405`)
+| The URI is valid, but the Method is not appropriate for the URI.
+
+  When `mask-rax-roles-403` attribute is enabled (i.e. `true`), this could also indicate the supplied `X-Roles` request header value had access to the URI, but not the requested Method.
+|===
```

Comment:
Are these all of the cases? There are fewer than I would have expected. What if the request payload does not validate against a schema? What if the request `Content-Type` does not match the expected `Content-Type`? What if the request `Accept` does not match the response `Content-Type`?

I'm pretty certain that a `406` response can occur.